core/unit_test/TestExecutionSpace.hpp

```diff
@@ -26,11 +26,15 @@ struct CheckClassWithExecutionSpaceAsDataMemberIsCopyable {
   Kokkos::DefaultHostExecutionSpace host;
 
   KOKKOS_FUNCTION void operator()(int, int& e) const {
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
     // not actually doing anything useful, mostly checking that
     // ExecutionSpace::in_parallel() is callable
     if (static_cast<int>(device.in_parallel()) < 0) {
       ++e;
     }
+#else
+    e = 0;
+#endif
```

Comment:
There is no point on keeping the deprecated code path.
```suggestion
    e += i;
```